### PR TITLE
Add a close button to the find bar. Fixes #1397. r=vporof

### LINKED ISSUE #1397: "Find" should have a close button
Especially since it only closes on `esc` if it has focus. I thought this was a bug until I realized Firefox does the same thing, and I guess I'm used to the close button.

### PR DESCRIPTION
**Pull request by jsantell**

### CODE REVIEW

File: app/ui/browser-modern/views/browser/decorations/searchbar.jsx [hunk 1]
```diff
@@ -16,22 +16,33 @@ import { connect } from 'react-redux';
...
+  backgroundColor: 'var(--theme-content-background)',
+  width: '300px',
+});
+
+const CLOSE_BUTTON_STYLE = Style.registerStyle({
+  margin: '6px 3px 0px 0px',
+});
+
+const SEARCH_INPUT_STYLE = Style.registerStyle({
+  width: '300px',
```
Comment (victorporof) [inline app/ui/browser-modern/views/browser/decorations/searchbar.jsx]: Why limit the search input width again here?

File: app/ui/browser-modern/views/browser/decorations/searchbar.jsx [hunk 2]
```diff
@@ -16,22 +16,33 @@ import { connect } from 'react-redux';
...
   zIndex: UIConstants.SEARCH_BAR_INDEX,
-  borderTop: '1px solid var(--theme-content-border-color)',
+  border: '1px solid var(--theme-content-border-color)',
   color: 'var(--theme-content-color)',
+  backgroundColor: 'var(--theme-content-background)',
+  width: '300px',
+});
+
+const CLOSE_BUTTON_STYLE = Style.registerStyle({
+  margin: '6px 3px 0px 0px',
```
Comment (victorporof) [inline app/ui/browser-modern/views/browser/decorations/searchbar.jsx]: This vertical centering via margins is hacky and bound to break. Please use correct flexbox rules.

File: app/ui/browser-modern/views/browser/decorations/searchbar.jsx [hunk 3]
```diff
@@ -16,22 +16,33 @@ import { connect } from 'react-redux';
...
   left: 0,
   right: 0,
   height: '25px',
   padding: '0 4px',
   zIndex: UIConstants.SEARCH_BAR_INDEX,
-  borderTop: '1px solid var(--theme-content-border-color)',
+  border: '1px solid var(--theme-content-border-color)',
   color: 'var(--theme-content-color)',
+  backgroundColor: 'var(--theme-content-background)',
+  width: '300px',
```
Comment (victorporof) [inline app/ui/browser-modern/views/browser/decorations/searchbar.jsx]: Why limit the width?

File: app/ui/browser-modern/views/browser/decorations/searchbar.jsx (hunk 2 above)
Comment (victorporof) [inline app/ui/browser-modern/views/browser/decorations/searchbar.jsx]: To be clear, this is the only thing I'm requesting changes for.

File: app/ui/browser-modern/views/browser/decorations/searchbar.jsx (hunk 3 above)
Comment (jsantell) [inline app/ui/browser-modern/views/browser/decorations/searchbar.jsx]: It's more obvious to me it's open at all when it's not full width, not a strong preference

File: app/ui/browser-modern/views/browser/decorations/searchbar.jsx (hunk 2 above)
Comment (jsantell) [inline app/ui/browser-modern/views/browser/decorations/searchbar.jsx]: Fixed via `align-items`

**Comment by jsantell** [comment]
All comments addressed!

**Pull request merged**